File: src/main/java/com/openvalue/boardgameratings/service/boardgame/BoardGameRepository.java
package com.openvalue.boardgameratings.service.boardgame;

import com.openvalue.boardgameratings.api.BoardGame;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface BoardGameRepository extends CrudRepository<BoardGameEntity, Long> {

    Optional<BoardGameEntity> findByName(String name);

    List<BoardGameEntity> findAll();
}

File: src/main/java/com/openvalue/boardgameratings/service/boardgame/BoardGameRepository.kt
package com.openvalue.boardgameratings.service.boardgame

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface BoardGameRepository : CrudRepository<BoardGameEntity, Long> {
    fun findByName(name: String): Optional<BoardGameEntity>

    override fun findAll(): List<BoardGameEntity>
}